src/settings_type.h

```diff
@@ -132,7 +132,8 @@ struct GUISettings {
 	bool   timetable_arrival_departure;      ///< show arrivals and departures in vehicle timetables
 	bool   right_mouse_wnd_close;            ///< close window with right click
 	bool   pause_on_newgame;                 ///< whether to start new games paused or not
-	bool   enable_signal_gui;                ///< show the signal GUI when the signal button is pressed
+	uint8  signal_gui_mode;                  ///< select which signal types are shown in the signal GUI
```

Comment:
I think these two could be their enum type instead of `uint8`. Some variables do it like that, but I also see others that are `uint8` (despite being an enum), so I am not 100% sure.

Reply:
When it's an enum use the enum type, agree. Perhaps put the storage type on the enum declaration: `enum SignalGUISettings : uint8 { ...`